compiler/rustc_errors/src/emitter.rs

```diff
@@ -2082,7 +2249,7 @@ impl HumanEmitter {
                             buffer.putc(
                                 row_num,
                                 (padding as isize + p) as usize,
-                                if part.is_addition(sm) { '+' } else { '~' },
+                                if part.is_addition(sm) { '+' } else { self.diff() },
```

Comment:
Side note: For `annotate-snippet`, it would be nice if symbols like `+` that currently don't vary by `OutputTheme` get moved to the "palette".